src/diag.rs

```diff
@@ -187,7 +187,7 @@ fn annotate_diagnostic(notes: &mut Vec<Notation>,
     }
 
     fn t(v: &Token) -> String {
-        String::from_utf8((**v).to_owned()).expect("utf-8 is checked before making Token")
+        unsafe { String::from_utf8_unchecked((**v).to_owned()) }
```

Comment:
Why this change?  Currently all of our unsafe blocks 1. locally check their preconditions 2. live in utils.rs 3. have a measurable performance impact; I'm especially concerned by the fact that this has a nonlocal dependency